Automattic/vip-design-system
# TS migration: Moves progress component to TS

## Description

Moves progress component to TS and adds a couple of basic tests.

## Checklist


## Steps to Test

We should test that the local storybook shows the same progress bar as the prod story book.
1. pull down the PR
2. run npm run dev
3. check story book component on localhost:6006 and compare with prod story book progress component — should be same.
